graphql_api/actions/commits.py

```diff
@@ -45,6 +46,34 @@ def commit_uploads(commit: Commit) -> QuerySet[ReportSession]:
     return sessions
 
 
+def commit_status(
+    commit: Commit, report_type: CommitReport.ReportType
+) -> Optional[CommitStatus]:
+    report = CommitReport.objects.filter(report_type=report_type, commit=commit).first()
+    if not report:
+        return None
+
+    sessions = report.sessions.all()
+    if not sessions:
+        return None
+
+    # Only care about these 3 states, ignoring fully and partially overwritten
+    upload_states = [
+        s.state for s in sessions if s.state in ["processed", "uploaded", "error"]
+    ]
+
+    # Returning commit status in order of priority
+    # If all are processed -> COMPLETED
+    if all(state == "processed" for state in upload_states):
```

Comment:
could combine this into a single pass if you want, not sure the typical size of states

return_status = CommitStatus.COMPLETED.value
for state in upload_states:
  if state == 'error':
    has_error = true
 if state == 'uploaded":
    has_pending = true

if has_error:
  return CommitStatus.ERROR.value
if has_pending:
  return CommitStatus.PENDING.value

At the very least, I don't think we need the first check since if any are error or uploaded we return early, and if not we would have gone through the whole list :)